packages/nimble-components/src/patterns/button/styles.ts

```diff
@@ -253,89 +264,139 @@ export const styles = css`
 );
 
 export const buttonAppearanceVariantStyles = css``.withBehaviors(
-    appearanceBehavior(
-        ButtonAppearance.outline,
-        css`
-            :host([appearance-variant='primary']) .control {
-                box-shadow: 0px 0px 0px ${borderWidth}
-                    rgba(${actionRgbPartialColor}, 0.3) inset;
-            }
-
-            :host([appearance-variant='primary']) .control:hover {
-                box-shadow: 0px 0px 0px ${borderWidth} ${borderHoverColor} inset;
-            }
-
-            :host([appearance-variant='primary']) .control${focusVisible} {
-                box-shadow: 0px 0px 0px ${borderWidth} ${borderHoverColor} inset;
-            }
-
-            :host([appearance-variant='primary']) .control:active {
-                box-shadow: 0px 0px 0px ${borderWidth} ${fillSelectedColor}
-                    inset;
-            }
-
-            :host([appearance-variant='primary'][disabled]) .control {
-                box-shadow: none;
-            }
-        `
+    appearanceVariantBehavior(
```

Comment:
Rather than nesting behaviors here, you might be able to simplify things by defining a CSS private variable for the color to use, and then conditionally changing the value of that variable for `[appearance-variant='primary'] or `[appearance-variant='accent']. See how the tooltip styling changes color variables for severities: https://github.com/ni/nimble/blob/bfd00b11c6f63fd57a7902a104627abbdc97195b/packages/nimble-components/src/tooltip/styles.ts#L70

Reply:
Yup, this works great! I added this in my newest commit, let me know what you think.

Reply:
This looks good, but there's a lot of duplication. What I was hoping you might be able to do (I haven't looked closely enough to be sure) is something like this:

```css
export const styles = css`
    ...
    .control {
         --ni-private-button-background-color: (some value for the default appearance variant)
         --ni-private-button-text-color: (some value for the default appearance variant)
         background-color: var(--ni-private-button-background-color);
         color: var(--ni-private-button-text-color);
    }
    ...
`;
export const buttonAppearanceVariantStyles = css`
     :host([appearance-variant='primary']) .control {
         --ni-private-button-background-color: ${buttonPrimaryBackgroundColor};
         --ni-private-button-text-color: ${buttonPrimaryFontColor};
     }
     :host([appearance-variant='accent']) .control {
         --ni-private-button-background-color: ${buttonAccentBackgroundColor};
         --ni-private-button-text-color: ${buttonAccentFontColor};
     }
     ...
`;
```
Basically, the idea would be to see if you can set most/all of the attributes using a few private variables that you declare, then just change the values of those variables based on the `appearance-variant` value. If I'm not explaining clearly, I can come by and chat about it.